x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -106,7 +106,7 @@ export const StepDetailsForm: FC<StepDetailsFormProps> = React.memo(
     const [createDataView, setCreateDataView] = useState(
       canCreateDataView === false ? false : defaults.createDataView
     );
-    const [dataViewAvailableTimeFields, setDataViewAvailableTimeFields] = useState<string[]>([]);
+    const [destIndexAvailableTimeFields, setDestIndexAvailableTimeFields] = useState<string[]>([]);
```

Comment:
Doesn't seem like we need a state for it. It's always derived from the `transformPreview`, hence we can replace it with `useMemo`.

Reply:
refactored in https://github.com/elastic/kibana/pull/172609/commits/9e983fb7171165b12cd668f4c04e0c2550ce952c